src/Parser/YaccParser.mly

```diff
@@ -153,26 +230,132 @@ expr
 | KW_EFFECT expr_250_list effect_resumption_opt ARROW2 expr
     { make (EEffect($2, $3, $5)) }
 | KW_HANDLER expr { make (EHandler $2) }
+| expr_0 { $1 }
+;
+
+expr_no_comma
+: def_list1 KW_IN expr_no_comma  { make (EDefs($1, $3)) }
+| KW_FN expr_250_list1 ARROW2 expr_no_comma { make (EFn($2, $4))   }
+| KW_EFFECT expr_250_list effect_resumption_opt ARROW2 expr_no_comma
+    { make (EEffect($2, $3, $5)) }
+| KW_HANDLER expr_no_comma { make (EHandler $2) }
+| expr_0_no_comma { $1 }
+;
+
+
+//Evaluate first expresion then discard the resulte and after that evaluate second expression
+//pattern = exp1 ; exp2
+expr_0
+: expr_10 op_0 expr {make (EBOp($1, $2, $3))}
 | expr_10 { $1 }
 ;
 
+
+expr_0_no_comma
+: expr_10_no_comma op_0 expr_no_comma {make (EBOp($1, $2, $3))}
+| expr_10_no_comma{ $1 }
+;
+
 effect_resumption_opt
 : /* empty */ { None    }
 | SLASH expr  { Some $2 }
 ;
 
 expr_10
 : KW_IF expr KW_THEN expr_10 KW_ELSE expr_10 { make (EIf($2, $4, $6)) }
+| expr_20 { $1 }
+;
+
+expr_10_no_comma
+: KW_IF expr_no_comma KW_THEN expr_10_no_comma KW_ELSE expr_10_no_comma{ make (EIf($2, $4, $6)) }
+| expr_20_no_comma { $1 }
+;
+
+//pattern = exp1 <- exp2
+expr_20
+: expr_30 op_20 expr_20 { make (EBOp($1,$2,$3)) }
 | expr_30 { $1 }
 ;
 
+expr_20_no_comma
+: expr_30_no_comma op_20 expr_20_no_comma { make (EBOp($1,$2,$3)) }
+| expr_30_no_comma { $1 }
+;
+
 expr_30
 : expr_30 COLON ty_expr { make (EAnnot($1, $3)) }
+| expr_35 { $1 }
+;
+
+expr_30_no_comma
+: expr_30_no_comma COLON ty_expr { make (EAnnot($1, $3)) }
+| expr_35_no_comma { $1 }
+;
+
+expr_35
+// pattern = exp1 , exp2
```

Comment:
I do not understand the role of `pattern` in this comment. We could remove these comments, or reformulate them to e.g., `expr1 , expr2`. I opt for the second options, as it would help connect operator precedence with precedence of other language constructs.